src/waveform/renderers/waveformmarkset.cpp

```diff
@@ -60,15 +65,27 @@ void WaveformMarkSet::setup(const QString& group, const QDomNode& node,
 
             if (controlItemSet.insert(hotCueControlItem).second) {
                 //qDebug() << "WaveformRenderMark::setup - Automatic mark" << hotCueControlItem;
-                m_marks.push_back(m_defaultMark);
-                WaveformMark& mark = m_marks.back();
-                mark.setKeyAndIndex(pHotcue->getKey(), i);
+                m_marks.push_back(new WaveformMark(m_pDefaultMark.get()));
+                WaveformMark* mark = m_marks.back();
+                mark->setKeyAndIndex(pHotcue->getKey(), i);
             }
         }
     }
 }
 
 void WaveformMarkSet::clear() {
-    m_defaultMark = WaveformMark();
+    m_pDefaultMark = std::make_unique<WaveformMark>();
+    qDeleteAll(m_marks);
     m_marks.clear();
 }
+
+const WaveformMark* WaveformMarkSet::getHotCueMark(int hotCue) {
+    DEBUG_ASSERT(hotCue >= 0);
+    DEBUG_ASSERT(hotCue < NUM_HOT_CUES);
+    return m_marks[m_iFirstHotCue + hotCue];
+}
+
+void WaveformMarkSet::setHotCueMark(int hotCue, WaveformMark* mark) {
```

Comment:
A plain pointer does not reveal the intention to transfer ownership. If you are using shared pointers for the elements of the set (see above comment) then pass in a shared pointer.

If you are in a situation where you explicitly need to transfer ownership for an object that has been allocated dynamically somewhere else then use std::unique_ptr passed by-value.